Source/FortranInterface/WarpX_picsar.F90

```diff
@@ -301,6 +314,58 @@ subroutine warpx_current_deposition( &
 
   end subroutine warpx_current_deposition
 
+  ! _________________________________________________________________
+  !>
+  !> @brief
+  !> Applies the inverse volume scaling for RZ current deposition
+  !>
+  !> @details
+  !> The scaling is done for both single mode (FDTD) and
+  !> multi mode (spectral)
+  !
+  !> @param[inout] jx,jy,jz current arrays
+  !> @param[in] nmodes number of spectral modes
+  !> @param[in] rmin tile grid minimum radius
+  !> @param[in] dr radial space discretization steps
+  !> @param[in] nx,ny,nz number of cells
+  !> @param[in] nxguard,nyguard,nzguard number of guard cells
+  !>
+  subroutine warpx_current_deposition_rz_volume_scaling( &
+    jx,jx_ng,jx_ntot,jy,jy_ng,jy_ntot,jz,jz_ng,jz_ntot,nmodes, &
+    rmin,dr) &
+    bind(C, name="warpx_current_deposition_rz_volume_scaling")
+
+    integer, intent(in) :: jx_ntot(AMREX_SPACEDIM), jy_ntot(AMREX_SPACEDIM), jz_ntot(AMREX_SPACEDIM)
+    integer(c_long), intent(in) :: jx_ng, jy_ng, jz_ng
+    integer(c_long), intent(IN) :: nmodes
+    real(amrex_real), intent(IN OUT):: jx(*), jy(*), jz(*)
+    real(amrex_real), intent(IN) :: rmin, dr
+
+    integer(c_long) :: type_rz_depose = 1
+
+    ! Compute the number of valid cells and guard cells
+    integer(c_long) :: jx_nvalid(AMREX_SPACEDIM), jy_nvalid(AMREX_SPACEDIM), jz_nvalid(AMREX_SPACEDIM), &
+                       jx_nguards(AMREX_SPACEDIM), jy_nguards(AMREX_SPACEDIM), jz_nguards(AMREX_SPACEDIM)
+    jx_nvalid = jx_ntot - 2*jx_ng
+    jy_nvalid = jy_ntot - 2*jy_ng
+    jz_nvalid = jz_ntot - 2*jz_ng
+    jx_nguards = jx_ng
+    jy_nguards = jy_ng
+    jz_nguards = jz_ng
+
+#ifdef WARPX_RZ
+    CALL WRPX_PXR_RZ_VOLUME_SCALING(   &
+                 jx,jx_nguards,jx_nvalid, &
+                 jy,jy_nguards,jy_nvalid, &
+                 jz,jz_nguards,jz_nvalid, &
+#ifdef WARPX_USE_PSATD
+                 nmodes, &
+#endif
```

Comment:
If `nmodes` does represent the number of azimuthal modes, then I am not sure it should be inside `#ifdef WARPX_USE_PSATD`. Since the underlying picsar function does not seem to support this parameter anyway, I would suggest to remove the whole block
```
#ifdef WARPX_USE_PSATD
                 nmodes, &
#endif
```
for now.